### PR TITLE
Remove and leave account feature

### PR DESCRIPTION
**Pull request by OscarSanner**
Adds methods for removing a contact from your contact list in the model package. Also adds feature to leave a group. These  features are added as methods in the Account class, which is delegated by the ModelEngine class. 

Changes to support this implementation also had to be added to the MockDatabase class.

### CODE REVIEW

File: app/src/main/java/com/goayo/debtify/model/MockDatabase.java [hunk 1]
```diff
@@ -110,15 +141,33 @@ public boolean registerGroup(String name, Set<User> users) {
...
+        for(User u : users){
+            if(u.getPhoneNumber().equals(phoneNumber)){
+                userToBeRemoved = u;
+            }
+        }
+
+        for (Group g : getGroups(phoneNumber)){
+            if(g.getGroupID().equals(groupID)){
+                if(userToBeRemoved != null){
+                    g.removeUser(userToBeRemoved); //Todo, wrong dependency order??
```
Comment (OscarSanner) [inline app/src/main/java/com/goayo/debtify/model/MockDatabase.java:158]: Ignore the comment

File: app/src/main/java/com/goayo/debtify/model/Account.java [hunk 2]
```diff
@@ -240,8 +269,8 @@ private Group getGroupFromID(String groupID) throws Exception {
         return database.getContactList(phoneNumber);
     }
 
-    private Set<Group> initAssociatedGroups(String phoneNumber) throws Exception {
-        return database.getGroups(phoneNumber);
+    private void loadAssociatedGroups() throws Exception {
+        associatedGroups = database.getGroups(loggedInUser.getPhoneNumber());
     }
 
     private void userIsLoggedIn() throws Exception {
```
Comment (yenanw) [inline app/src/main/java/com/goayo/debtify/model/Account.java:276]: not relevant to this commit but, shouldn't this just return a boolean?

Comment (yenanw) [reply app/src/main/java/com/goayo/debtify/model/Account.java:276]: or do we just assume the user has to be logged in no matter what?

**Review by yenanw — APPROVED** [review]

**Review by OlofSjogren — APPROVED** [review]
Nicely added, @yenanw will be handled in the next refactor.

**Pull request merged**